lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+        git_fetch
+      end
+
+      def create_cross_repo_test_branch
+        rugged_repo.create_branch(branch_name, "origin/master")
+        rugged_repo.checkout(branch_name)
+      end
+
+      def update_travis_yaml_content
+        Dir.chdir(@rugged_repo.workdir) do
+          content = YAML.load_file(".travis.yml")
+
+          content["env"] = {} unless content["env"]
+          content["env"]["global"] = ["REPOS=#{repos.join(',')}"]
+          content["env"]["matrix"] = test_repos.map { |repo| "TEST_REPO=#{repo}" }
+
+          File.write('.travis.yml', content.to_yaml)
+        end
+      end
+
+      def commit_travis_yaml_changes
+        index = rugged_repo.index
+        index.add('.travis.yml')
+        index.write
+
+        # rubocop:disable: Rails/TimeZone
+        bot       = self.class.bot_name
+        author    = { :name => issuer, :email => "no-name@example.com", :time => Time.now }
+        committer = { :name => bot,    :email => "#{bot}@manageiq.org", :time => Time.now }
+        # rubocop:enable: Rails/TimeZone
+
+        Rugged::Commit.create(
+          rugged_repo,
+          :author     => author,
+          :committer  => committer,
+          :parents    => [rugged_repo.last_commit].compact,
+          :tree       => index.write_tree(rugged_repo),
+          :update_ref => "HEAD",
+          :message    => <<~COMMIT_MSG
+            Running tests for #{issuer}
+
+            From Pull Request:  #{issue.fq_repo_name}##{issue.number}
+          COMMIT_MSG
+        )
+      end
+
+      def push_commit_to_remote
+        push_options = {}
+
+        if Settings.github_credentials.username && Settings.github_credentials.password
+          rugged_creds = Rugged::Credentials::UserPassword.new(
+            :username => Settings.github_credentials.username,
+            :password => Settings.github_credentials.password
+          )
+          push_options[:credentials] = rugged_creds
+        end
+
+        remote = @rugged_repo.remotes['origin']
+        remote.push(["refs/heads/#{branch_name}"], push_options)
+      end
+
+      def create_cross_repo_test_pull_request
+        fq_repo_name = "#{issue.organization_name}/#{File.basename(self.class.test_repo_url, '.*')}"
+        pr_desc      = <<~PULL_REQUEST_MSG
+          From Pull Request:  #{issue.fq_repo_name}##{issue.number}
+          For User:           @#{issuer}
+        PULL_REQUEST_MSG
+
+        GithubService.create_pull_request(fq_repo_name,
+                                          "master", branch_name,
+                                          "[BOT] Cross repo test run", pr_desc)
+      end
+
+      ##### Duplicate Git stuffs #####
+
+      # Code that probably should be refactored to be shared elsewhere, but for
+      # now just shoving it here to get a working prototype together.
+
+      # Mostly a dupulicate from Repo.git_fetch (app/models/repo.rb)
+      #
+      # Don't need the credentials stuff since we are assuming https for this repo
```

Comment:
I think this is a wrong assumption.

Reply:
Talked about above, so going to resolve this for now.  Undo if you disagree.

Reply:
Marking this one as unresolved for now just so we can track it that it's not been done.

Reply:
As [mentioned above](https://github.com/ManageIQ/miq_bot/pull/454#discussion_r359090068), I will open up a issue once we merge this PR for handling this.  Once done, I will then resolve this again (basically, after merge though).